pkg/controller/dynamicrequiredmanifest_controller_test.go

```diff
@@ -161,3 +162,97 @@ func TestDynamicRequiredManifestControllerSync(t *testing.T) {
 		})
 	}
 }
+
+func TestUnstructuredShouldUpdateFunc(t *testing.T) {
+	for _, tc := range []struct {
+		name         string
+		manifest     []byte
+		existing     runtime.Object
+		expectUpdate bool
+		assertError  func(t *testing.T, err error)
+	}{
+		{
+			name:         "existing is nil, no error, update needed",
+			existing:     nil,
+			expectUpdate: true,
+			assertError:  noError(),
+		},
+		{
+			name:         "existing is not *unstructured.Unstructured, error",
+			existing:     &corev1.Pod{},
+			expectUpdate: false,
+			assertError:  containsError(errors.New("expected existing to be of type *unstructured.Unstructured but was")),
+		},
+		{
+			name:     "required and existing are not deep derivative, no error, update needed",
+			manifest: []byte(requiredYAML),
+			existing: &unstructured.Unstructured{
+				Object: map[string]interface{}{
+					"apiVersion": "olm.operatorframework.io/v1alpha1",
+					"kind":       "ClusterCatalog",
+					"metadata": map[string]interface{}{
+						"name": "openshift-certified-operators",
+					},
+					"spec": map[string]interface{}{
+						"source": map[string]interface{}{
+							"type": "Image",
+							"image": map[string]interface{}{
+								"pollInterval": "1h",
+								"ref":          "foobarbaz",
+							},
+						},
+					},
+				},
+			},
+			expectUpdate: true,
+			assertError:  noError(),
+		},
+		{
+			name:     "required and existing are deep derivative, no error, no update needed",
+			manifest: []byte(requiredYAML),
+			existing: &unstructured.Unstructured{
```

Comment:
Add a `metadata.labels` map with some custom values and set `Availability: Disabled` and make sure we still see "no updated needed"?

Reply:
https://github.com/openshift/cluster-olm-operator/pull/73/commits/b90b97c06be0d80e94126fd55109fa049daa75a1